addons/stock/models/stock_move.py

```diff
@@ -387,8 +387,24 @@ def name_get(self):
                 move.location_id.name, move.location_dest_id.name)))
         return res
 
+    def set_move_name_in_vals(self, vals):
```

Comment:
Have you tried with a api.depends storable field, to redefine the field name on stock move that depends on product_id

Reply:
The problem is we want to avoid the `name` field being computed on the fly when adding a new move. If I added `api.depends` then this would be triggered each time a product was selected.

But to be honest this change only had a marginal improvement on the time taken to add new lines, so it will probably be dropped. I will go check what we want to do this.

Reply:
Spoke with Miquel and we're gonna drop this change. :)

Reply:
Dropped. :)